File: parcial.py
class Libro:
    def __init__(self, titulo, autor, año, editorial, ISBN, paginas, stock):
        self.titulo = titulo
        self.autor = autor
        self.año = año
        self.editorial = editorial
        self.ISBN = ISBN
        self.paginas = paginas
        self.stock = stock

class Lector:
    def __init__(self, nombre, DNI, libro_solicitado):
        self.nombre = nombre
        self.DNI = DNI
        self.libro_solicitado = libro_solicitado

def guardar_libro(archivo, libro):
    archivo.write(f"Titulo: {libro.titulo}\n")
    archivo.write(f"Autor: {libro.autor}\n")
    archivo.write(f"Se publico en: {libro.año}\n")
    archivo.write(f"Editorial: {libro.editorial}\n")
    archivo.write(f"ISBN: {libro.ISBN}\n")
    archivo.write(f"Cantidad de paginas: {libro.paginas}\n")
    archivo.write(f"Stock disponible: {libro.stock}\n")
    archivo.write("\n")

def nuevo_libro(archivo):
    titulo = input("Ingrese el título: ")
    autor = input("Ingrese el autor: ")
    año = int(input("Ingrese el año de publicación: "))
    editorial = input("Ingrese la editorial: ")
    ISBN = input("Ingrese el ISBN: ")
    paginas = int(input("Ingrese la cantidad de páginas: "))
    stock = int(input("Ingrese el stock disponible: "))
    libro = Libro(titulo, autor, año, editorial, ISBN, paginas, stock)
    guardar_libro(archivo, libro)

def main():
    try:
        with open("biblioteca.txt", "a") as archivo:
            print("Hola, bienvenido al gestionador de biblioteca. Por favor, elija la operación que desea realizar escribiendo el número que le corresponde y presionando ENTER.")
            print("1. Agregar libros a la lista enlazada.")
            print("2. Ordenar libros.")
            print("3. Buscar libros.")
            print("4. Solicitar libros.")
            print("5. Devolver libros.")
            print("6. Guardar y cargar datos.")
            opcion = int(input())

            if opcion == 1:
                print("Ha elegido la opción 1.")
                nuevo_libro(archivo)
                print("El archivo se ha actualizado correctamente. Fin de la ejecución.")
            elif opcion == 2:
                print("Ha elegido la opción 2.")

                # print("Los libros se han ordenado alfabéticamente en biblioteca.txt.")
            elif opcion == 3:
                print("Ha elegido la opción 3.")
 
            elif opcion == 4:
                print("Ha elegido la opción 4.")

            elif opcion == 5:
                print("Ha elegido la opción 5.")

            elif opcion == 6:
                print("Ha elegido la opción 6.")

            else:
                print("Por favor, elija una opción válida.")

    except IOError:
        print("Error, no se pudo abrir el archivo biblioteca.txt")

if __name__ == "__main__":
    main()

File: parcial.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

struct libro {
    string titulo;
    string autor;
    int año;
    string editorial;
    string ISBN;
    int paginas;
    int stock;
};

struct lector {
    string nombre; 
    int DNI;
    string libro_solicitado;
};

void guardarlibro(ofstream& archivo, const libro& libro) {
    archivo << "Título: " << libro.titulo << endl;
    archivo << "Autor: " << libro.autor << endl;
    archivo << "Año de publicación: " << libro.año << endl;
    archivo << "Editorial: " << libro.editorial << endl;
    archivo << "ISBN: " << libro.ISBN << endl;
    archivo << "Cantidad de páginas: " << libro.paginas << endl;
    archivo << "Stock disponible: " << libro.stock << endl;
    archivo << endl;
}

void nuevolibro(ofstream& archivo) {
    libro nuevolibro;

    cout << "Ingrese los datos del libro: " << endl;
    cout << "Titulo: ";
    cin.ignore();
    getline(cin, nuevolibro.titulo);

    cout << "Autor: ";
    getline(cin, nuevolibro.autor);

    cout << "Año de publicacion: ";
    cin >> nuevolibro.año;

    cout << "Editorial: ";
    cin.ignore(); 
    getline(cin, nuevolibro.editorial);

    cout << "ISBN: ";
    cin.ignore();
    getline(cin, nuevolibro.ISBN);

    cout << "Cantidad de paginas: ";
    cin >> nuevolibro.paginas;

    cout << "Stock disponible: ";
    cin >> nuevolibro.stock;

    guardarlibro(archivo, nuevolibro);
}

int main() {
    int opcion;
//Crea el archivo biblioteca.txt
    ofstream libros("biblioteca.txt", ios::app);
    if (libros.fail()) {
        cout << "Error, no se pudo abrir el archivo biblioteca.txt" << endl;
        exit(1);
    }

    cout << "Hola, bienvenido al gestionador de biblioteca. Porfavor, elija que operacion desea realizar escribiendo el numero que le corresponde y presionando ENTER." << endl;

    cout << "1. Agregar libros a la lista enlazada." << endl;
    cout << "2. Ordenar libros." << endl;
    cout << "3. Buscar libros." << endl;
    cout << "4. Solicitar libros." << endl;
    cout << "5. Devolver libros." << endl;
    cin >> opcion;

    switch(opcion) {
        case 1:
            cout << "Ha elegido la opcion 1." << endl;
            nuevolibro(libros);
            cout << "El archivo se ha actualizado correctamente. Fin de la ejecucion.";
            break;
        
        case 2:
            cout << "Ha elegido la opcion 2." << endl;
            //cout << "El archivo se ha actualizado correctamente. Fin de la ejecucion.";
        break;

        case 3:
            cout << "Ha elegido la opcion 3." << endl;
            //cout << "El archivo se ha actualizado correctamente. Fin de la ejecucion.";
        break;

        case 4:
            cout << "Ha elegido la opcion 4." << endl;
            //cout << "El archivo se ha actualizado correctamente. Fin de la ejecucion.";
        break;

        case 5:
            cout << "Ha elegido la opcion 5." << endl;
            //cout << "El archivo se ha actualizado correctamente. Fin de la ejecucion.";
        break;

        default:
            cout << "Porfavor, elija una opcion valida." << endl;
        break;
    }

    return 0;
} 